search_commands.go

```diff
@@ -1954,7 +1954,7 @@ func (c cmdable) FTSearchWithArgs(ctx context.Context, index string, query strin
 				args = append(args, "WITHCOUNT")
 			}
 		}
-		if options.LimitOffset >= 0 && options.Limit > 0 {
```

Comment:
If both the `LimitOffset` and `Limit` are left to the default `0` values, we will end up with `LIMIT 0 0` which is described as a special usecase of the limit, to return count of the results instead of the results themselves (search for `LIMIT 0 0` in the docs: https://redis.io/docs/latest/commands/ft.search/ )